### PR TITLE
Separate semaphore target into a clean target for use on other CI systems

### PR DESCRIPTION
**Pull request by jose-bigio**
Signed-off-by: jose-bigio <jose.bigio@docker.com>

## Description

- **Fix type**: Continuous Integration

The CI system utilizes the `semaphore` target to build. This target first cleans the Semaphore machine with: 
```
bash -c 'rm -rf /usr/local/golang /opt /var/lib/mongodb /usr/lib/jvm /home/runner/{.npm,.phpbrew,.phpunit,.kerl,.kiex,.lein,.nvm,.npm,.phpbrew,.rbenv}'
```

It would be desirable to utilize this target on other CI systems. For instance, I want to set up a different CI on a fork, but running the clean target would be destructive to the CI.

This change would require a change to the  semaphore CI, because the bash commands would need to be updated to call `make clean-semaphore` and `make semaphore` as opposed to only `make semaphore`. 

On a related note I wanted to know if the CI commands such as 
```
if [ -e calico_node/Makefile ]; then if [ -z $PULL_REQUEST_NUMBER ] || git diff --name-only HEAD^ | grep '^calico_node/' || git diff --name-only HEAD^ | grep '^_data/versions.yml'; then if grep batchnumber calico_node/tests/st/test_base.py; then sudo -E ST_OPTIONS="-A batchnumber==0" make -C calico_node semaphore; else sudo -E make -C calico_node semaphore; fi; fi; fi
```
are stored in source control anywhere? 

## Todos

- [ ] ~Update Semaphore CI to call make clean-semaphore && make semaphore~
- [ ] Documentation (If the CI system is documented anywhere possibly update the process there)

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
> are stored in source control anywhere?

I don't believe they are.

@jose-bigio what do you think about factoring it something like this:

```
.PHONY: ci
ci:
 <run the tests>

.PHONY: semaphore
semaphore: clean-semaphore ci
```

This way, we don't need to change the meaning of the `semaphore` target, and we provide a more generic CI target that other systems can run.

**Comment by caseydavenport** [comment]
> Error processing tar file(exit status 1): write /bin/confd: no space left on device

I'm guessing the current semaphore failures are from disk space issues introduced by this PR.

**Comment by jose-bigio** [comment]
Hey @caseydavenport  thanks for the suggestion if it looks good I can squash or delete my fist commit.

If it's desirable I would like to add the bash scripts that are used to set up Semaphore as part of a Readme, so the process is in source control in some way.  (This can be separate PR).

Also in the semaphore job I see that there are some batch jobs run off of the `test.sh` script. 
I can't find this in the repo. 
Is this stored in semaphore, or is it generated by something?

File: calico_node/Makefile [hunk 1]
```diff
@@ -588,6 +588,17 @@ node-test-at:
 	chmod +rx /tmp/goss && \
 	/tmp/goss --gossfile /tmp/goss.yaml validate'
 
+.PHONY: ci 
+ci:
```
Comment (caseydavenport) [inline calico_node/Makefile]: This is the same as most of the semaphore target below.

Why don't we make the semaphore target like this:

```
semaphore:
  <do cleanup>
  $(MAKE) ci
```

**Review by caseydavenport — APPROVED** [review]
LGTM!

**Pull request merged**